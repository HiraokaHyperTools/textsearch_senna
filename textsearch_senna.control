# textsearch_senna extension
comment = 'textsearch_senna'
default_version = '9.0.1'
module_pathname = '$libdir/textsearch_senna'
relocatable = false
