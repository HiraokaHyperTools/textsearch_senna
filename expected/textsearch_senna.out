\set ECHO all
SET client_encoding = utf8;
SET client_min_messages = notice;
SELECT senna.contains('あいう', senna.to_tsquery('あい'));
 contains 
----------
 t
(1 row)

SELECT senna.contains('あいう', senna.to_tsquery('いう'));
 contains 
----------
 t
(1 row)

SELECT senna.contains('あいう', senna.to_tsquery('え'));
 contains 
----------
 f
(1 row)

SELECT 'あいう' %% 'あい';
 ?column? 
----------
 t
(1 row)

SELECT 'あいう' @@ 'あい'::senquery;
 ?column? 
----------
 t
(1 row)

SELECT 'あいう' @@ senna.to_tsquery('あい');
 ?column? 
----------
 t
(1 row)

SELECT senna.to_tsvector('あいう') @@ senna.to_tsquery('あい');
 ?column? 
----------
 t
(1 row)

SELECT senna.to_tsvector('english', 'あいう') @@ senna.to_tsquery('english', 'あい');
 ?column? 
----------
 t
(1 row)

SELECT sum(CASE WHEN 'あい1う' %% ('あい' || (i % 4)) THEN 1 ELSE 0 END)  FROM generate_series(1, 1000) t(i);
 sum 
-----
 250
(1 row)

SELECT sum(CASE WHEN 'あい1う' %% ('あい' || (i % 100)) THEN 1 ELSE 0 END)  FROM generate_series(1, 1000) t(i);
 sum 
-----
  10
(1 row)

CREATE TABLE tbl (id int4, t text);
INSERT INTO tbl VALUES(1, 'あいう');
INSERT INTO tbl VALUES(2, 'いうえ');
INSERT INTO tbl VALUES(3, 'うえお');
INSERT INTO tbl VALUES(10, 'ABC');
INSERT INTO tbl VALUES(11, 'BCD');
INSERT INTO tbl VALUES(12, 'CDE');
INSERT INTO tbl VALUES(20, 'A B');
INSERT INTO tbl VALUES(21, E'A \\ B');
INSERT INTO tbl VALUES(22, 'A " B');
INSERT INTO tbl VALUES(23, 'A * B');
INSERT INTO tbl VALUES(24, 'A + B');
INSERT INTO tbl VALUES(25, 'A % B');
INSERT INTO tbl VALUES(26, 'A _ B');
CREATE INDEX idx_senna ON tbl USING senna (t) WITH (initial_n_segments = 4);
CREATE INDEX idx_like ON tbl USING senna (t like_ops) WITH (initial_n_segments = 2);
SELECT relname, reloptions FROM pg_class WHERE relname IN ('idx_senna', 'idx_like') ORDER BY relname;
  relname  |       reloptions       
-----------+------------------------
 idx_like  | {initial_n_segments=2}
 idx_senna | {initial_n_segments=4}
(2 rows)

SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
  2 | いうえ
(2 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
  2 | いうえ
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
 11 | BCD
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
  2 | いうえ
(2 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
  2 | いうえ
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
 11 | BCD
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
  2 | いうえ
(2 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
  2 | いうえ
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
 11 | BCD
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

INSERT INTO tbl VALUES(4, 'あいうえお');
INSERT INTO tbl VALUES(5, 'かきくけこ');
INSERT INTO tbl VALUES(14, 'ABCDE');
INSERT INTO tbl VALUES(15, 'abcde');
SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id |     t      
----+------------
  1 | あいう
  2 | いうえ
  4 | あいうえお
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id |     t      
----+------------
  1 | あいう
  2 | いうえ
  4 | あいうえお
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id |     t      
----+------------
  1 | あいう
  2 | いうえ
  4 | あいうえお
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id |     t      
----+------------
  1 | あいう
  2 | いうえ
  4 | あいうえお
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id |     t      
----+------------
  1 | あいう
  2 | いうえ
  4 | あいうえお
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id |     t      
----+------------
  1 | あいう
  2 | いうえ
  4 | あいうえお
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id |   t    
----+--------
  1 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

UPDATE tbl SET id = 999 WHERE id = 1;
UPDATE tbl SET t = 'さしす' WHERE id = 2;
UPDATE tbl SET t = 'いういう' WHERE id = 3;
UPDATE tbl SET t = 'AB CD' WHERE id = 12;
UPDATE tbl SET t = 'BCBC' WHERE id = 13;
SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%B C%' ORDER BY id;
 id |   t   
----+-------
 12 | AB CD
(1 row)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%B C%' ORDER BY id;
 id |   t   
----+-------
 12 | AB CD
(1 row)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%B C%' ORDER BY id;
 id |   t   
----+-------
 12 | AB CD
(1 row)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

VACUUM ANALYZE tbl;
SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   3 | いういう
   4 | あいうえお
 999 | あいう
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

DELETE FROM tbl WHERE id = 3;
SELECT senna.reindex_index('idx_senna');
 reindex_index 
---------------
 
(1 row)

SELECT senna.reindex_index('idx_like');
 reindex_index 
---------------
 
(1 row)

SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   4 | あいうえお
 999 | あいう
(2 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   4 | あいうえお
 999 | あいう
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   4 | あいうえお
 999 | あいう
(2 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   4 | あいうえお
 999 | あいう
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;
SELECT * FROM tbl WHERE t %% '不在' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t %% 'いう' ORDER BY id;
 id  |     t      
-----+------------
   4 | あいうえお
 999 | あいう
(2 rows)

SELECT * FROM tbl WHERE t ~~% '%不在%' ORDER BY id;
 id | t 
----+---
(0 rows)

SELECT * FROM tbl WHERE t ~~% '%いう%' ORDER BY id;
 id  |     t      
-----+------------
   4 | あいうえお
 999 | あいう
(2 rows)

SELECT * FROM tbl WHERE t ~~% 'あ%う' ORDER BY id;
 id  |   t    
-----+--------
 999 | あいう
(1 row)

SELECT * FROM tbl WHERE t ~~% '%BC%' ORDER BY id;
 id |   t   
----+-------
 10 | ABC
 11 | BCD
 14 | ABCDE
(3 rows)

SELECT * FROM tbl WHERE t ~~% 'A%C' ORDER BY id;
 id |  t  
----+-----
 10 | ABC
(1 row)

SELECT count(*) FROM tbl WHERE t ~~% '%いう%'
UNION ALL
SELECT count(*) FROM tbl WHERE t ~~% '%いう%';
 count 
-------
     2
     2
(2 rows)

SELECT * FROM tbl WHERE t ~~% '%あい%' AND t ~~% '%お%';
 id |     t      
----+------------
  4 | あいうえお
(1 row)

SELECT * FROM tbl WHERE t ~~% '%A B%';
 id |  t  
----+-----
 20 | A B
(1 row)

SELECT * FROM tbl WHERE t ~~% E'%A\\ B%';
 id |  t  
----+-----
 20 | A B
(1 row)

SELECT * FROM tbl WHERE t ~~% E'%A \\\\ B%';
 id |   t   
----+-------
 21 | A \ B
(1 row)

SELECT * FROM tbl WHERE t ~~% '%A " B%';
 id |   t   
----+-------
 22 | A " B
(1 row)

SELECT * FROM tbl WHERE t ~~% '%A * B%';
 id |   t   
----+-------
 23 | A * B
(1 row)

SELECT * FROM tbl WHERE t ~~% '%A + B%';
 id |   t   
----+-------
 24 | A + B
(1 row)

SELECT * FROM tbl WHERE t ~~% E'%A \\% B%';
 id |   t   
----+-------
 25 | A % B
(1 row)

SELECT * FROM tbl WHERE t ~~% E'%A \\_ B%';
 id |   t   
----+-------
 26 | A _ B
(1 row)

-- search for empty keys
SELECT * FROM tbl WHERE t ~~% '';
ERROR:  scan key must not be empty for senna index: public.idx_like
SELECT * FROM tbl WHERE t ~~% '%';
ERROR:  scan key must not be empty for senna index: public.idx_like
SELECT * FROM tbl WHERE t ~~% '_';
ERROR:  scan key must not be empty for senna index: public.idx_like
SELECT * FROM tbl WHERE t ~~% '%%';
ERROR:  scan key must not be empty for senna index: public.idx_like
SELECT * FROM tbl WHERE t ~~% '%_%';
ERROR:  scan key must not be empty for senna index: public.idx_like
SELECT * FROM tbl WHERE t ~~% '"';
ERROR:  scan key must not be empty for senna index: public.idx_like
SELECT * FROM tbl WHERE t ~~% E'\\';
ERROR:  LIKE pattern must not end with escape character
SELECT * FROM tbl WHERE t ~~% NULL;
 id | t 
----+---
(0 rows)

SELECT senna.checkpoint();
 checkpoint 
------------
 
(1 row)

SELECT senna.drop_index('idx_senna');
 drop_index 
------------
 
(1 row)

SELECT senna.drop_index('idx_like');
 drop_index 
------------
 
(1 row)

CHECKPOINT;
SELECT * FROM senna.orphan_files;
 file 
------
(0 rows)

